src/main/java/com/microsoft/azure/util/AzureMsiCredentials.java

```diff
@@ -2,24 +2,87 @@
 
 import com.cloudbees.plugins.credentials.CredentialsScope;
 import com.cloudbees.plugins.credentials.impl.BaseStandardCredentials;
+import com.microsoft.azure.AzureEnvironment;
 import hudson.Extension;
+import hudson.util.ListBoxModel;
+import org.apache.commons.lang.StringUtils;
 import org.kohsuke.stapler.DataBoundConstructor;
 
+import java.util.Map;
+
 public class AzureMsiCredentials extends BaseStandardCredentials {
 
     public static final int DEFAULT_MSI_PORT = 50342;
+    private static final long serialVersionUID = 6743945878507124459L;

```

Comment:
I would suggest we use a human readable ID (e.g., `1L`) and increase the number when we have incompatible fields change.